src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Declarations.cs

```diff
@@ -85,6 +86,7 @@ private SearchQuery(Func<string, bool> predicate)
                 throw new ArgumentNullException(nameof(predicate));
             }
 
+            Kind = SearchKind.Custom;
```

Comment:
We could, yes.  However, note that supplying a custom predicate is still allowed through SymbolFinder.  It will just never flow into the SymbolTreeInfo codepath.  That custom predicate codepath will do htings like flow into the compiler functions for searching for symbols.